zhmccli/_cmd_partition.py

```diff
@@ -916,6 +967,133 @@ def cmd_partition_console(cmd_ctx, cpc_name, partition_name, options):
+                if _sg.name == sg_name:
+                    sg = _sg
+                    break
+            if not sg:
+                raise_click_exception(
+                    "Storage group '{sg}' specified in the --volume option was "
+                    "not found in the storage groups attached to partition "
+                    "'{part}' of CPC '{cpc}'.".
+                    format(sg=sg_name, part=partition_name, cpc=cpc_name),
+                    cmd_ctx.error_format)
+            try:
+                storage_volume = sg.storage_volumes.find(name=sv_name)
+            except zhmcclient.NotFound:
+                raise_click_exception(
+                    "Storage volume '{sv}' specified in the --volume option "
+                    "was not found in the volumes of storage group '{sg}' "
+                    "attached to partition '{part}' of CPC '{cpc}'.".
+                    format(sv=sv_name, sg=sg_name, part=partition_name,
+                           cpc=cpc_name),
+                    cmd_ctx.error_format)
+        else:
+            raise_click_exception(
+                "Invalid format for --volume option: '{v}'. CPC '{cpc}' has "
+                "the storage management feature which requires the formats "
+                "'SG.SV' or 'UUID'.".
+                format(v=volume_opt, cpc=cpc_name),
+                cmd_ctx.error_format)
+
+        # Set the remaining feature-specific parameters
+        properties['storage-volume-uri'] = storage_volume.uri
+        properties['timeout'] = timeout
+
+        op_properties = {
+            'dump-program-info': properties,
+            'dump-program-type': 'storage',
+        }
+        try:
+            partition.start_dump_program(
+                parameters=op_properties, wait_for_completion=True)
+        except zhmcclient.Error as exc:
+            raise_click_exception(exc, cmd_ctx.error_format)
+
+    else:  # No storage management feature
+
+        # Check and parse the --volume option into HBA.WWPN.LUN
+        volume_parts = volume_opt.split('.')
+        if len(volume_parts) != 3:
+            raise_click_exception(
+                "Invalid format for --volume option: '{v}'. CPC '{cpc}' does "
+                "not have the storage management feature which requires the "
+                "format 'HBA.WWPN.LUN'.".
+                format(v=volume_opt, cpc=cpc_name),
+                cmd_ctx.error_format)
+        hba_name, wwpn, lun = volume_parts
+        try:
+            hba = partition.hbas.find(name=hba_name)
+        except zhmcclient.NotFound:
+            raise_click_exception(
+                "HBA '{hba}' specified in the --volume option was not found "
+                "in partition '{part}' of CPC '{cpc}'.".
+                format(hba=hba_name, part=partition_name, cpc=cpc_name),
+                cmd_ctx.error_format)
+
+        # Set the remaining feature-specific parameters
+        properties['dump-load-hba-uri'] = hba.uri
```

Comment:
Is it okay that we reuse the properties dict here?

Reply:
We are not reusing the variable name for a different purpose, we continue setting it up for the same purpose as above. I have renamed it to "parameters" as part pf the splitting of the two functions.

Reply:
AFAICR, the properties dict was used for two calls.

Reply:
Not in the current code.